plugins/tweak/tweaks/embark-profile-name.h

```diff
@@ -1,22 +1,21 @@
 #include "df/viewscreen_setupdwarfgamest.h"
-using namespace DFHack;
+
 struct embark_profile_name_hook : df::viewscreen_setupdwarfgamest {
     typedef df::viewscreen_setupdwarfgamest interpose_base;
-    DEFINE_VMETHOD_INTERPOSE(void, feed, (std::set<df::interface_key> *input))
-    {
+
+    DEFINE_VMETHOD_INTERPOSE( void, feed, ( std::set<df::interface_key> *input ) ) {
         int ch = -1;
-        for (auto it = input->begin(); ch == -1 && it != input->end(); ++it)
-            ch = Screen::keyToChar(*it);
-        if (in_save_profile && ch >= 32 && ch <= 126)
-        {
-            profile_name.push_back((char)ch);
+        for( auto it = input->begin(); ch == -1 && it != input->end(); ++it ) {
+            ch = Screen::keyToChar( *it );
         }
-        else
-        {
-            if (input->count(df::interface_key::LEAVESCREEN))
```

Comment:
I may not have been clear about this on IRC - while the esc key wasn't mentioned on-screen, I would prefer to keep it because it's a conventional key to leave a screen/dialog (so people will tend to press it), and it doesn't do anything in this particular dialog otherwise (so supporting it won't conflict with DF)